src/katsdpmodels/primary_beam.py

```diff
@@ -234,7 +323,437 @@ def sample_grid(self, l: ArrayLike, m: ArrayLike, frequency: u.Quantity,   # noq
         Refer to :meth:`sample` for further details of the parameters and
```

Comment:
The method abstract suggests the grid must be regular, but the extended abstract contradicts this.